src/Client/Aws/SnsClient.php

```diff
@@ -70,18 +75,42 @@
  */
 class SnsClient extends AbstractAwsClient
 {
-    const RESOURCE_NAME = 'TopicArn';
+    protected const RESOURCE_NAME = 'TopicArn';
+
+    private const SnsMessage = 'Message';
 
     /**
      * @param mixed $message
      *
      * @return Result
      */
-    public function send($message)
+    public function send($message): Result
+    {
+        return $this->publish([self::SnsMessage => json_encode($message),]);
+    }
+
+    /**
+     * @param MessageInterface $message
+     * @return Result
+     * @throws \Exception
+     */
+    public function sendMessage(MessageInterface $message): Result
+    {
+        return $this->sendEnvelope(new Envelope($message));
+    }
+
+    /**
+     * @param Envelope $messageEnvelop
+     * @return Result
+     * @throws \Exception
+     */
+    public function sendEnvelope(Envelope $messageEnvelop): Result
     {
-        $args = ['Message' => json_encode($message)];
+        $messageEnvelop->setCreatedAt(new \DateTimeImmutable());
 
-        return $this->publish($args);
+        return $this->publish([
+            self::SnsMessage => $this->serializer->serialize($messageEnvelop, JsonEncoder::FORMAT)
+        ]);
```

Comment:
Should we join these 2 functions since the `sendMessage()` has become just a wrapper?

Reply:
Not really. We keep the future-proof factor, in case you want to have a custom envelope.

Reply:
I'm not sure it's worth it in this case, since we don't know if we are ever going to need a custom envelope and until that time it will make the code a little bit harder to read...

I would say it's such a simple change to implement that it can be inserted back in the future if needed  :)

Reply:
I have to disagree. Since this is a separate repo from the services, people are less incentivized to contribute to it. As it is now there no harm, just two methods of sending messages, on for convenience, one for utility.

Reply:
Alright :+1: